packages/backend/src/open_payments/auth/middleware.ts

```diff
@@ -33,39 +38,54 @@ export function createAuthMiddleware({
         return
       }
       const authService = await ctx.container.use('authService')
-      const grant = await authService.introspect(token)
-      if (!grant || !grant.active) {
+      const tokenInfo = await authService.introspect(token)
+      if (!tokenInfo || !tokenInfo.active) {
         ctx.throw(401, 'Invalid Token')
       }
       if (
-        !grant.includesAccess({
+        !tokenInfo.includesAccess({
           type,
           action,
           identifier: ctx.paymentPointer.url
         })
       ) {
         ctx.throw(403, 'Insufficient Grant')
       }
+      try {
+        const sig = stringifyHeader(ctx.headers['signature'])
+        const sigInput = stringifyHeader(ctx.headers['signature-input'])
+        const successfullyVerified = await verifySig(
+          sig.match(/:([^:]+):/)[1],
+          tokenInfo.key.jwk,
+          sigInput
```

Comment:
This argument needs to be the be output of [`sigInputToChallenge`](https://github.com/interledger/rafiki/blob/0f0ff4e2f40e042a315d8c85889b4f71d37d451a/packages/auth/src/signature/middleware.ts#L89).
Maybe you could instead export and use [`verifySigAndChallenge`](https://github.com/interledger/rafiki/blob/0f0ff4e2f40e042a315d8c85889b4f71d37d451a/packages/auth/src/signature/middleware.ts#L20).
in which case, you might need to change [`HttpSigHeaders`](https://github.com/interledger/rafiki/blob/0f0ff4e2f40e042a315d8c85889b4f71d37d451a/packages/auth/src/signature/middleware.ts#L119-L141), etc to use `Koa.Context` instead of auth's `AppContext`